source/opt/trim_capabilities_pass.cpp

```diff
@@ -159,46 +159,54 @@ TrimCapabilitiesPass::TrimCapabilitiesPass()
 void TrimCapabilitiesPass::addInstructionRequirements(
     Instruction* instruction, CapabilitySet* capabilities,
     ExtensionSet* extensions) const {
+  // Ignoring OpCapability instructions.
+  if (instruction->opcode() == spv::Op::OpCapability) {
+    return;
+  }
+
   // First case: the opcode is itself gated by a capability.
-  {
+  do {
     const spv_opcode_desc_t* desc = {};
     auto result =
         context()->grammar().lookupOpcode(instruction->opcode(), &desc);
-    if (result == SPV_SUCCESS) {
-      addSupportedCapabilitiesToSet(desc->numCapabilities, desc->capabilities,
-                                    capabilities);
+    if (result != SPV_SUCCESS) {
+      break;
     }
 
+    addSupportedCapabilitiesToSet(desc->numCapabilities, desc->capabilities,
+                                  capabilities);
     if (desc->minVersion <= spvVersionForTargetEnv(context()->GetTargetEnv())) {
       extensions->insert(desc->extensions,
                          desc->extensions + desc->numExtensions);
     }
-  }
+  } while (0);
```

Comment:
It is hard to follow the control flow. Adding the single-iteration loop so you can break is on par with using a goto.  Try

```suggest
  const spv_opcode_desc_t* desc = {};
  auto result =
      context()->grammar().lookupOpcode(instruction->opcode(), &desc);
    if (result == SPV_SUCCESS) {
      addSupportedCapabilitiesToSet(desc->numCapabilities, desc->capabilities,
                                  capabilities);
      if (desc->minVersion <= spvVersionForTargetEnv(context()->GetTargetEnv())) {
        extensions->insert(desc->extensions,
                           desc->extensions + desc->numExtensions);
      }
    }
```